app/bibleview-js/src/components/modals/AskBookmarkSettings.vue

```diff
@@ -0,0 +1,805 @@
+import ModalDialog from "@/components/modals/ModalDialog.vue";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { Deferred } from "@/utils";
+import { useCommon } from "@/composables";
+import { faTimes, faBan, faPlus, faArrowUp, faExchangeAlt, faIcons, faEdit, faParagraph, faHeading, faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+import { customIconMap } from "@/composables/fontawesome";
+import { EditAction, EditActionMode } from "@/types/client-objects";
+
+const { strings } = useCommon();
+
+interface BookmarkSettings {
+    customIcon: string | null;
+    editAction: EditAction;
+}
+
+const show = ref(false);
+const activeTab = ref<'icons' | 'editAction'>('icons');
+const selectedIcon = ref<null | string>(null);
+const selectedEditAction = reactive<EditAction>({
+    mode: null,
+    content: null
+});
+const contentTextarea = ref<HTMLTextAreaElement | null>(null);
+const validationError = ref<string | null>(null);
+
+let deferred: Deferred<BookmarkSettings | null> | null = null;
+
+function selectIcon(key: null | string) {
+    selectedIcon.value = key;
+}
+
+function insertParagraphBreak() {
+    const textarea = contentTextarea.value;
+    if (!textarea) return;
+    
+    const start = textarea.selectionStart;
+    const end = textarea.selectionEnd;
+    const currentValue = selectedEditAction.content || '';
+    
+    // Insert <br/> at cursor position
+    const beforeCursor = currentValue.substring(0, start);
+    const afterCursor = currentValue.substring(end);
+    const newValue = beforeCursor + '<br/>' + afterCursor;
+    
+    selectedEditAction.content = newValue;
+    
+    // Move cursor after the inserted tag
+    setTimeout(() => {
+        textarea.selectionStart = textarea.selectionEnd = start + 5;
+        textarea.focus();
+    }, 0);
+}
+
+function insertSubtitle() {
+    const textarea = contentTextarea.value;
+    if (!textarea) return;
+    
+    const start = textarea.selectionStart;
+    const end = textarea.selectionEnd;
+    const currentValue = selectedEditAction.content || '';
+    
+    // Get selected text or use placeholder
+    const selectedText = currentValue.substring(start, end);
+    const subtitleText = selectedText || 'Subtitle text';
+    
+    // Insert <subtitle>...</subtitle> at cursor position
+    const beforeCursor = currentValue.substring(0, start);
+    const afterCursor = currentValue.substring(end);
+    const newValue = beforeCursor + `<subtitle>${subtitleText}</subtitle>` + afterCursor;
+    
+    selectedEditAction.content = newValue;
+    
+    // Select the subtitle text for easy editing
+    setTimeout(() => {
+        const subtitleStart = start + 10; // after '<subtitle>'
+        const subtitleEnd = subtitleStart + subtitleText.length;
+        textarea.selectionStart = subtitleStart;
+        textarea.selectionEnd = subtitleEnd;
+        textarea.focus();
+    }, 0);
+}
+
+function validateXmlContent(content: string): string | null {
```

Comment:
this function should be moved to somewhere (utility), generalized a bit (allow tags to be general list of tags), and then unit-tested properly